### ISSUE TITLE
Row details displays undefined rows

### ISSUE THREAD

**Issue by crlsmsq** [ghiscoding/Angular-Slickgrid#1092]
### Describe the bug

Hi! I tried implementing the detailed row feature. But clicking a row only shows new lines with undefined values
![image](https://user-images.githubusercontent.com/80941041/216247533-08550501-0aa9-4882-a9ac-e2ddf049faa1.png)

Checking the click event of the plugin shows that I got 0 `container_*` classes 
![image](https://user-images.githubusercontent.com/80941041/216247681-66f92ff9-4a2d-4e64-abfc-3b788dc858f5.png)

I was wondering if I could reinitialize the plugin? I have a dynamic dataset that changes from one to the other with different set of columns and rows using the same slickgrid instance. 

I tried the .invalidate() but it doesn't reinitialize the plugins. Also tried this function i found, but it relies on a different class with the same name i think. 
![image](https://user-images.githubusercontent.com/80941041/216248351-ef7dbb12-c329-4440-9de8-3efa9fd3d07b.png)

### Reproduction

I followed the tutorial on how to use it. i think it happens because after the slickgrid was initialized with no column headers and data. i gave the user an option to generate the data with dynamic columns and apply it in slickgrid

### Environment Info

```shell
angular 13
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/angular-slickgrid/wiki/HOWTO---Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/Angular-Slickgrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/Angular-Slickgrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
> I followed the tutorial on how to use it. i think it happens because after the slickgrid was initialized with no column headers and data. i gave the user an option to generate the data with dynamic columns and apply it in slickgrid

it was never created to be used that way and is not supported either, you also can't register/unregister that way because it doesn't use the regular SlickGrid registering process and that simply won't work. I thought the following code might work
```ts
  angularGridReady(angularGrid: AngularGridInstance) {
    this.angularGrid = angularGrid;
  }

  get rowDetailInstance(): SlickRowDetailView {
    // you can get the SlickGrid RowDetail plugin (addon) instance via 2 ways

    // option 1
    return (this.angularGrid.extensions.rowDetailView.instance || {});

    // OR option 2
    // return this.angularGrid?.extensionService.getExtensionInstanceByName(ExtensionName.rowDetailView) || {};
  }

  later() {
    this.rowDetailInstance.dispose(); // to unsubscrive events & avoid mem leaks
    this.rowDetailInstance.init(this.angularGrid.slickGrid); // reinitialize the plugin
  }
```
but it doesn't work and is causing different error (Maximum call stack size exceeded). Let's put it this way, Row Detail is created once and only once and if you're creating the columns dynamically in a later process, I think you're looking for trouble. I don't have more time to spend on this and so at this point I would say Row Detail is not supported dynamically

**Comment by crlsmsq** [comment]
hi ghiscoding! thanks for this. okay I'll just create the slickgrid instance when I have the columns and rows for it :)

**Comment by ghiscoding** [comment]
but really what's the point of initializing a grid without columns? Why don't you just use an `ngIf` to create your grid when you have all available data at hand, using the `ngIf` would certainly work correctly

**Comment by crlsmsq** [comment]
yes ill do that :D

**Comment by crlsmsq** [comment]
hello @ghiscoding ! 

Applying a column preset from the slickgrid config makes these `+` expandible button missing.

w/o applying preset
![image](https://user-images.githubusercontent.com/80941041/226853927-36b03154-5cfb-4376-8da4-c9c15aa7f931.png)


with applying preset
![image](https://user-images.githubusercontent.com/80941041/226854735-23f5faf7-6b98-4280-aaad-946359429146.png)

**Comment by ghiscoding** [comment]
Sorry but why are you reopening this issue when the new issue you are describing is completely different from the original? Please don't hack old issue just to save time when it's totally different. 

You are also missing a lot of details that I require for troubleshooting
- you did not mention your Angular-Slickgrid version (note I only support latest version).
   -  I don't think this issue is happening with the latest version, I remember seeing something similar in the past. 
- you also did not provide any code to reproduce.

**Comment by crlsmsq** [comment]
thank you so much for this

**Comment by ghiscoding** [comment]
you can thank me by staring ⭐ the lib if you haven't already 😉

**Comment by crlsmsq** [comment]
yes <3 i just did
